### ISSUE TITLE
futures-io: "error: cannot find macro `poll_fill_bufderef_async_buf_read` in this scope"

### ISSUE THREAD

**Issue by Tpt** [async-rs/async-std#755]
Thank you very much for this so nice library!

When trying to run `cargo check` on a clean clone of async-std with Rust stable 1.43.0 on Linux 5.6 I end up with the following error from futures-io:

```
error: cannot find macro `poll_fill_bufderef_async_buf_read` in this scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/futures-io-0.3.4/src/lib.rs:547:9
    |
547 |         poll_fill_bufderef_async_buf_read!();
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0046]: not all trait items implemented, missing: `poll_fill_buf`, `consume`
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/futures-io-0.3.4/src/lib.rs:546:5
    |
305 |         fn poll_fill_buf(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<&[u8]>>;
    |         ------------------------------------------------------------------------------------ `poll_fill_buf` from trait
...
322 |         fn consume(self: Pin<&mut Self>, amt: usize);
    |         --------------------------------------------- `consume` from trait
...
546 |     impl<T: ?Sized + AsyncBufRead + Unpin> AsyncBufRead for &mut T {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `poll_fill_buf`, `consume` in implementation
```

I have not managed to reproduce this error on the current master of `futures-rs` git repository. This error affects both async-std master and version 1.4.

I am not familiar with async-std and futures internals so I have not been able to bet a good understanding of the possible cause of this problem.

**Comment by taiki-e** [comment]
Since there is no macro named `poll_fill_bufderef_async_buf_read` in [published futures-io 0.3.4](https://docs.rs/futures-io/0.3.4/src/futures_io/lib.rs.html#509), it is likely that the local `.cargo/registry/src/github.com-1ecc6299db9ec823/futures-io-0.3.4` was accidentally rewritten.

e.g., by using "jump to the definition" or similar features access the source code of dependencies from the editor, then replace one of the `deref_async_buf_read` macro calls with `poll_fill_bufderef_async_buf_read` (I could reproduce this error by this way). 
Also, when correcting, replace `poll_fill_bufderef_async_buf_read` with `deref_async_buf_read` in the same way.

**Comment by Tpt** [comment]
Indeed. It's exactly that. Sorry for the dumb bug report.
